src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,80 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange = () => {},
+}: Props) => {
+  const lastPage = Math.ceil(total / perPage);
+
+  return (
+    <ul className="pagination">
+      <li
+        className={classNames('page-item', {
+          disabled: currentPage === 1,
+        })}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          onClick={() => {
+            if (currentPage > 1) {
+              onPageChange(currentPage - 1);
+            }
+          }}
+          aria-disabled={currentPage === 1 ? 'true' : 'false'}
+        >
+          «
+        </a>
+      </li>
+
+      {getNumbers(1, lastPage).map(number => (
+        <li
+          key={number}
+          className={classNames('page-item', {
+            active: currentPage === number,
+          })}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${number}`}
```

Comment:
Same as the previous comment, using `href="#${number}"` can affect the URL state. It's better to use `event.preventDefault()` to avoid changing the URL when these links are clicked.